modules/modal/react/lib/Modal.tsx

```diff
@@ -69,125 +69,124 @@ const transformOrigin = {
+      elem.focus();
+    }
+  }, [modalRef]);
+};
+
+const Modal = ({
+  open,
+  handleClose,
+  padding,
+  width,
+  heading,
+  children,
+  closeOnEscape,
+  firstFocusRef,
+  ...elemProps
+}: ModalProps): JSX.Element | null => {
+  const modalRef = React.useRef<HTMLDivElement>(null);
+
+  const handleKeydown = (event: KeyboardEvent) => {
+    if (modalRef.current) {
+      tabTrappingKey(event, modalRef.current);
+    }
 
-  private handleKeydown = (event: KeyboardEvent) => {
-    if (
-      this.props.closeOnEscape &&
-      this.props.handleClose &&
-      (event.key === 'Esc' || event.key === 'Escape')
-    ) {
-      this.props.handleClose();
+    if (closeOnEscape && handleClose && (event.key === 'Esc' || event.key === 'Escape')) {
+      handleClose();
     }
   };
 
-  componentDidMount() {
-    document.addEventListener('keydown', this.handleKeydown);
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleKeydown);
-  }
+  useKeyDownListener(handleKeydown);
+  useInitialFocus(modalRef, firstFocusRef);
 
-  private handleOutsideClick = (event: React.MouseEvent<HTMLDivElement>) => {
-    const {target} = event;
-    const modalNode = this.modalRef.current;
-    if (modalNode && this.props.handleClose) {
-      if (!modalNode.contains(target as Node)) {
-        this.props.handleClose();
-      }
+  const handleOutsideClick = ({target}: React.MouseEvent<HTMLDivElement>) => {
+    const modalEl = modalRef.current;
+    if (modalEl && !modalEl.contains(target as Node) && handleClose) {
+      handleClose();
     }
   };
 
-  public render() {
-    const {
-      open,
-      handleClose,
-      padding,
-      width,
-      heading,
-      children,
-      closeOnEscape,
-      firstFocusRef: firstFocusableRef,
-      ...elemProps
-    } = this.props;
-
+  if (open) {
     return (
-      open && (
-        <FocusTrap
-          focusTrapOptions={{
-            initialFocus: () =>
-              onInitialFocus(this.modalRef.current, firstFocusableRef && firstFocusableRef.current),
-          }}
+      <Container onClick={handleOutsideClick} {...elemProps}>
+        <Popup
+          popupRef={modalRef}
+          width={width}
+          heading={heading}
+          handleClose={handleClose}
+          padding={padding}
+          transformOrigin={transformOrigin}
         >
-          <Container onClick={this.handleOutsideClick} {...elemProps}>
-            <Popup
-              popupRef={this.modalRef}
-              width={width}
-              heading={heading}
-              handleClose={handleClose}
-              padding={padding}
-              transformOrigin={transformOrigin}
-            >
-              {children}
-            </Popup>
-          </Container>
-        </FocusTrap>
-      )
+          {children}
+        </Popup>
+      </Container>
     );
   }
-}
+  return null;
+};
+
+Modal.Padding = PopupPadding;
+Modal.Width = ModalWidth;
+
+Modal.propTypes = {
```

Comment:
Did you mean `defaultProps`? That might be why the interface had to be changed.

Reply:
@NicholasBoll thanks! that was very silly mistake :)